pulpcore/cli/common/generic.py

```diff
@@ -435,3 +456,155 @@ def label_show(entity_ctx: PulpEntityContext, key: str) -> None:
         label_group.add_command(subcmd)
 
     return label_group
+
+
+def repository_content_command(**kwargs: Any) -> click.Group:
+    """A factory that creates a repository content command group."""
+
+    if "name" not in kwargs:
+        kwargs["name"] = "content"
+    content_contexts = kwargs.pop("contexts", {})
+    content_names = list(content_contexts.keys())
+    decorators = kwargs.pop("decorators", [href_option])
+
+    @click.group(**kwargs)
+    @click.option(
+        "--type",
+        "-t",
+        "content_type",
+        type=click.Choice(content_names, case_sensitive=False),
+        default=content_names[0],
+    )
+    @pass_pulp_context
+    @click.pass_context
+    def content_group(ctx: click.Context, pulp_ctx: PulpContext, content_type: str) -> None:
+        ctx.obj = content_contexts[content_type](pulp_ctx)
+        subcommand = ctx.command.commands[ctx.invoked_subcommand]  # type: ignore
+        if ctx.invoked_subcommand in ["add", "remove"]:
+            for field in ctx.obj.UNIQUE_LOOKUPS:
+                click.option(
+                    f"--{field.replace('_', '-')}",
+                    help=_(field.capitalize() + " of the {entity} to add to the repository"),
```

Comment:
I think I tried this already and it didn't work.  The help message shows fine for me, but maybe it won't translate properly.

Reply:
Yes, that is the reason. The _() is interpreted in two ways:
1. When running the code in python, it will just look for whatever string it gets in the lookuptable.
2. When doing the string extraction with gettext. This will ignore everything that is not a literal string.
So the result will be that your string never shows up in translations, and will therefore be untranslatable.